### PR TITLE
Implement all sigs, quotes

### PR DESCRIPTION
**Pull request by cadaeic**
Additional changes
- Improve custom mega ability
- Allow unlimited mega evolution
- Fix Beowulf custom
- Prevent infinite chaining in all custom moves based on Protect-like moves
- Fix Enguarde custom
- Improve Jin custom
- Implement Relados new custom
- Nerf Tempoanon custom
- Nerf SOMALIA custom
- Add placeholder names for shrang and eos custom
- Add HappyFunTimes quotes
- Fix Cathy custom

### CODE REVIEW

**Comment by cadaeic** [comment]
I understand that I made some balance changes to existing sets. If there's a problem with any of these changes I guess they can be reverted after merging? 
@Joimer you're mon's receiving a lot of complaints
@Ascriptmaster I bypassed the mega limit part by directly using pokemon.getItem(). If you have the time maybe you can see into how the more regular canMegaEvo() way can be made to work?

File: config/formats.js [hunk 1]
```diff
@@ -2369,8 +2398,35 @@ exports.Formats = [
...
+				}
+				if (name === 'dtc' && (pokemon.getAbility().id === 'magicguard' || pokemon.getAbility().id === 'toughclaws')) {
+					pokemon.setAbility('levitate');
+				}
+				if (name === 'shrang' && pokemon.getAbility().id === 'levitate') {
+					pokemon.setAbility('pixilate');
+				}
+				if (name === 'trinitrotoluene' && (pokemon.getAbility().id === 'levitate' || pokemon.getAbility().id === 'toughclaws')) {
+					pokemon.setAbility('protean');
+				}
```
Comment (ascriptmaster) [inline config/formats.js]: When something goes mega their ability changes, so this is only needed when the requested ability of the mega does NOT match the mega's original ability. This means Slayer95 and Dell don't need their ability changes, and I think the others only need to check for the mega's original ability and not the base form's abilities.

Also you might need to check `target` as well, or else DTC has a chance of taking a Ground move without immunity if it moves second, and Shrang might dodge a Ground move for the same reason!

File: config/formats.js [hunk 2]
```diff
@@ -2980,7 +3068,7 @@ exports.Formats = [
 					pokemon.addVolatile('firespin');
 				};
 			}
-			if (move.id === 'allyswitch' && name === 'slayer95') {
+			if (move.id === 'allyswitch' && name === 'slayer95') {	//seems to be not fully functional
```
Comment (ascriptmaster) [inline config/formats.js]: What is actually happening here is that `name` becomes the illusion's name if you're behind an illusion. (See the beginning of this `onModifyMove` function.)

File: config/formats.js (hunk 1 above)
Comment (TheImmortal) [inline config/formats.js]: >I think the others only need to check for the mega's original ability and not the base form's abilities
You're right, I didn't think about that

Comment (cadaeic) [inline config/formats.js]: @ascriptmaster I've tested the ground immunities and they already work as intended.

**Comment by cadaeic** [comment]
@Joimer @TheImmortal I've fixed the mega abilities part too. As for Slayer95's custom, I still can't really understand what the code does so I guess I'll leave it to you guys.

**Comment by TheImmortal** [comment]
I don't think the onBeforeMove hack is needed with the onUpdate hack

**Comment by cadaeic** [comment]
Without the onBeforeMove hack Audiosurfer's Pixilate won't activate when he uses Boomburst right after mega evolving, for example

**Comment by cadaeic** [comment]
@TheImmortal I just tested the mega abilities again and it looks like you're right. I removed the onBeforeMove hack.

**Comment by cadaeic** [comment]
A new pull has been made instead.

**Pull request closed without merging**